tedge_config/src/settings.rs

```diff
@@ -126,6 +127,20 @@ impl ConfigSetting for AzureRootCertPathSetting {
     type Value = FilePath;
 }
 
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub struct MosquittoPortSetting;
+
+impl ConfigSetting for MosquittoPortSetting {
+    const KEY: &'static str = "mosquitto.port";
```

Comment:
This property should be `mqtt.port`.

Thin-edge is currently using mosquitto but is not tied to.